### PR TITLE
update DiscordJS

### LINKED ISSUE #151: Upgrade Discord.JS from v11 to v12
Discord will be introducing breaking changes that will conflict with DiscordJS v11.
Screen Shot 2021-03-31 at 5 03 25 PM

We will need to:

- Identify all areas that are affected by the upgrade
- Transition it over to v12 compliant syntax
- Update bot documentation where necessary
- Update tests (specifically, mocks) where necessary

### PR DESCRIPTION
**Pull request by twalton83**
- [X] You have read the [Odin Bot README/Contributing Guide](https://github.com/TheOdinProject/odin-bot-v2/blob/master/README.md).

#### 1.Describe the changes made:

Our DiscordJS v11 setup was going to become quickly obsolete as Discord is slated to roll out breaking changes any day now. This PR:

- updates the version to the stable, version 12
- updates areas where we referenced Collections directly, and now uses `.cache` property on those collections so we may continue to use Collection methods

#### 2. If this PR modifies a command, please provide a screenshot of the passing tests below. NOTE: Your PR will not be merged if you have any failing test cases. 

... Image here

#### 3. If this PR is related to an open issue please reference it with the `#` operator and the issue number below:

Closes #151

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1,7493 +1,8 @@
 {
   "name": "bot-v2",
   "version": "1.0.0",
-  "lockfileVersion": 2,
+  "lockfileVersion": 1,
   "requires": true,
-  "packages": {
```
Comment (KevinMulhern) [inline package-lock.json:6]: There seems to be a lot of deletions here for a  version bump. When you run npm install locally does it have any un-staged changes in the package-lock.json?

**Review by KevinMulhern — COMMENTED** [review]
Hey @twalton83, thanks for taking care of this. Its great to get ahead of the breaking changes 😄 

Just a couple of questions.

File: package-lock.json (hunk 1 above)
Comment (twalton83) [reply package-lock.json:6]: I just ran `npm i` while in my branch and it says it's up to date. I did notice that I don't have a "packages" key, but I have a "dependencies" key now.

File: botCommands/points.js [hunk 2]
```diff
@@ -140,7 +140,7 @@ const points = {
   }) {
     const userIds = getUserIdsFromMessage(content, /<@!?(\d+)>/g);
     userIds.forEach(async (userId) => {
-      const user = await client.users.get(userId);
+      const user = await client.users.cache.get(userId);
```
[the remark this answers is not in the document]
Comment (twalton83) [reply botCommands/points.js:143]: I did just test it, I did have the order wrong in a few areas but this does work *now*! We don't have to fetch it first in this case.

**Review by KevinMulhern — APPROVED** [review]
Looks great to me, thanks for all your work on this @twalton83 🚀

**Pull request merged**